src/sinks/appsignal/mod.rs

```diff
@@ -150,61 +163,279 @@ impl SinkConfig for AppsignalSinkConfig {
+    batch_service:
+        HttpBatchService<Ready<Result<http::Request<Bytes>, crate::Error>>, AppsignalRequest>,
+}
 
-    match response.status() {
-        status if status.is_success() => Ok(()),
-        other => Err(super::HealthcheckError::UnexpectedStatus { status: other }.into()),
+impl AppsignalService {
+    pub fn new(
+        http_client: HttpClient<Body>,
+        endpoint: Uri,
+        push_api_key: SensitiveString,
+        compression: Compression,
+    ) -> Self {
+        let batch_service = HttpBatchService::new(http_client, move |req| {
+            let req: AppsignalRequest = req;
+
+            let mut request = Request::post(&endpoint)
+                .header("Content-Type", "application/json")
+                .header(AUTHORIZATION, format!("Bearer {}", push_api_key.inner()))
+                .header("Content-Length", req.payload.len());
+            if let Some(ce) = compression.content_encoding() {
+                request = request.header("Content-Encoding", ce)
+            }
+            let result = request.body(req.payload).map_err(|x| x.into());
+            future::ready(result)
+        });
+        Self { batch_service }
+    }
+}
+
+impl Service<AppsignalRequest> for AppsignalService {
+    type Response = AppsignalResponse;
+    type Error = crate::Error;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(
+        &mut self,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, mut request: AppsignalRequest) -> Self::Future {
+        let mut http_service = self.batch_service.clone();
+
+        Box::pin(async move {
+            let metadata = std::mem::take(request.metadata_mut());
+            http_service.ready().await?;
+            let bytes_sent = metadata.request_wire_size();
+            let event_byte_size = metadata.into_events_estimated_json_encoded_byte_size();
+            let http_response = http_service.call(request).await?;
+            let event_status = if http_response.is_successful() {
+                EventStatus::Delivered
+            } else if http_response.is_transient() {
+                EventStatus::Errored
+            } else {
+                EventStatus::Rejected
+            };
+            Ok(AppsignalResponse {
+                event_status,
+                http_status: http_response.status(),
+                event_byte_size,
+                bytes_sent,
+            })
+        })
+    }
+}
+
+struct AppsignalResponse {
+    event_status: EventStatus,
+    http_status: StatusCode,
+    event_byte_size: GroupedCountByteSize,
+    bytes_sent: usize,
+}
+
+impl DriverResponse for AppsignalResponse {
+    fn event_status(&self) -> EventStatus {
+        self.event_status
+    }
+
+    fn events_sent(&self) -> &GroupedCountByteSize {
+        &self.event_byte_size
+    }
+
+    fn bytes_sent(&self) -> Option<usize> {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Incidentally, for correctness at this point in time you didn't need to add this (because the BytesSent isn't being set as a Registered event in the driver, and the internal metric is emitted as `EndpointBytesSent` below) 
However, I would recommend keeping this change as is because I believe we will want to use this trait function implementation when we make the EndpointBytesSent a registered event in the driver.